src/FastRoute.php

```diff
@@ -118,7 +118,24 @@ public function useCustomDispatcher(string $dataGenerator, string $dispatcher):
             $this->routeParser,
             $dataGenerator,
             $dispatcher,
-            $this->routesConfiguration,
+            $this->configureRoutes,
+            $this->uriGenerator,
+            $this->cacheDriver,
+            $this->cacheKey,
+        );
+    }
+
+    /**
+     * @param class-string<ConfigureRoutes> $configureRoutes
+     */
+    public function useCustomConfigureRoutes(string $configureRoutes): self
```

Comment:
Having an instance that implements should also be helpful

Reply:
That instance would be the FakeRouteCollector, referenced above.

Reply:
Sorry, I was multi-tasking and communicated poorly.

I meant to say having the ability to receive an instance. Your point on #280 makes a lot of sense, so we could postpone changes and enhance the design using closures in a separate pr.

Reply:
> we could postpone changes and enhance the design using closures in a separate pr

I have posted an alternative solution that incorporates something like "real" DI instead at #282 -- let me know if you think that path (or some variation of it) is viable.